bcipy/tasks/rsvp/query_mechanisms.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+from copy import copy
+import random
+
+# the small epsilon value. Prevents numerical issues with log
+eps = np.power(.1, 6)
+
+
+class StimuliAgent:
+    """ Query mechanism base class.
+        Attr:
+            alphabet(list[str]): Query space(possible queries).
+            len_query(int): number of elements in a query
+        Functions:
+            reset(): reset the agent
+            return_stimuli(): update the agent and return a stimuli set
+            do_epoch(): one a commitment is made update agent
+            """
+
+    def __init__(self, alphabet, len_query, **kwargs):
+        self.alp = alphabet
+        self.len_query = len_query
+
+    def reset(self):
+        return
+
+    def return_stimuli(self, list_distribution, **kwargs):
+        """ updates the agent with most likely posterior and selects queries
+            Args:
+                list_distribution(list[ndarray]): posterior distributions as
+                    stored in the decision maker
+            Return:
+                query(list[str]): queries """
+        return
+
+    def do_epoch(self):
+        """ If the system decides on a class let the agent know about it """
+        return
+
+
+class RandomStimuliAgent(StimuliAgent):
```

Comment:
We should have class documentation here.

Reply:
Class documentation added.